### PR TITLE
move client public ip check to use server

### PR DESCRIPTION
**Pull request by afeiszli**
This commit adds a public ip checking service to the netmaker server. By doing so, clients can request their public IP from the netmaker server, instead of reaching out to an external service. 

The purpose of this is twofold:
1. It enables the 0.0.0.0/0 use case by having a reliable endpoint for all clients to reach to get their public IP which can be added to the routing table.
2. This should actually enable Netmaker to work in several countries where it does not currently, without needing the "additional allowedips" functionality

### CODE REVIEW

**Comment by afeiszli** [comment]
To test (@theguy951357), spin up Netmaker with some clients. Check to see if they get the correct endpoints (should be fairly obvious if it is not working). Also, should test a client on the same server as Netmaker for safety, as well as clients that are in the same subnet (e.g. DigitalOcean region).

File: controllers/ipservice.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+		return "", err
+	}
+	netIP = net.ParseIP(ip)
+	if netIP != nil {
+		if netIP.IsPrivate() || netIP.IsLoopback() {
+			return ip, fmt.Errorf("ip is a private address")
+		}
+		return ip, nil
+	}
+	return "", fmt.Errorf("No valid ip found")
```
Comment (mattkasun) [inline controllers/ipservice.go:67]: as per the static check --- change to 'no valid ip found'

**Review by mattkasun — CHANGES_REQUESTED** [review]
fix static check erros

File: controllers/ipservice.go (hunk 1 above)
Comment (afeiszli) [reply controllers/ipservice.go]: changed to "no ip found" on both

**Review by 0xdcarns — APPROVED** [review]

**Review by mattkasun — CHANGES_REQUESTED** [review]
script changes should be in separate PR.

**Review by mattkasun — APPROVED** [review]

**Comment by mattkasun** [comment]
@theguy951357 RFT

**Review by theguy951357 — APPROVED** [review]
ips and ports look consistent. pass

**Pull request merged**